File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/03.Conditional-Statements-Advanced/03-additional/07.schoolCamp.js
function schoolCamp(input) {

    const overnightPriceGirlsOrBoysForWinter = 9.60;
    const overnightPriceGirlsOrBoysForSpring = 7.20;
    const overnightPriceGirlsOrBoysForSummer = 15.00;
    const overnightPriceMixedGroupForWinter = 10.00;
    const overnightPriceMixedGroupForSpring = 9.50;
    const overnightPriceMixedGroupForSummer = 20.00;

    // спортовете, които ще се практикуват в зависимост от вида на ваканцията и групата:
    const girlsWinter = "Gymnastics";
    const girlsSpring = "Athletics";
    const girlsSummer = "Volleyball";
    const boysWinter = "Judo";
    const boysSpring = "Tennis";
    const boysSummer = "Football";
    const mixedGroupWinter = "Ski";
    const mixedGroupSpring = "Cycling";
    const mixedGroupSummer = "Swimming";

    const season = input[0];
    const group = input[1];
    const studentsCount = Number(input[2]);
    const overnightCount = Number(input[3]);

    let price = studentsCount * overnightCount;
    let sport = "";

    switch (group) {
        case "boys":
            switch (season) {
                case "Winter": sport = boysWinter;
                    price *= overnightPriceGirlsOrBoysForWinter; break;
                case "Spring": sport = boysSpring;
                    price *= overnightPriceGirlsOrBoysForSpring; break;
                case "Summer": sport = boysSummer;
                    price *= overnightPriceGirlsOrBoysForSummer; break;
            }
            break;
        case "girls":
            switch (season) {
                case "Winter": sport = girlsWinter;
                    price *= overnightPriceGirlsOrBoysForWinter; break;
                case "Spring": sport = girlsSpring;
                    price *= overnightPriceGirlsOrBoysForSpring; break;
                case "Summer": sport = girlsSummer;
                    price *= overnightPriceGirlsOrBoysForSummer; break;
            }
            break;
        case "mixed":
            switch (season) {
                case "Winter": sport = mixedGroupWinter;
                    price *= overnightPriceMixedGroupForWinter; break;
                case "Spring": sport = mixedGroupSpring;
                    price *= overnightPriceMixedGroupForSpring; break;
                case "Summer": sport = mixedGroupSummer;
                    price *= overnightPriceMixedGroupForSummer; break;
            }
            break;
    }
    // • Ако броят на учениците е 50 или повече, училището получава 50% отстъпка
    // • Ако броят на учениците е 20 или повече и в същото време по-малък от 50, училището получава 15% отстъпка
    // • Ако броят на учениците е 10 или повече и в същото време по-малък от 20, училището получава 5% отстъпка
    if (studentsCount >= 50) {
        price *= 0.5;
    } else if (studentsCount >= 20) {
        price *= 0.85;
    } else if (studentsCount >= 10) {
        price *= 0.95;
    }
    console.log(`${sport} ${price.toFixed(2)} lv.`);
}
schoolCamp(["Summer", "boys", "60", "7"]);

File: Python_Basics/Homeworks-And-Labs/03.conditional_statements_advanced/03.additional/07.school_camp.py
overnightPriceGirlsOrBoysForWinter = 9.60
overnightPriceGirlsOrBoysForSpring = 7.20
overnightPriceGirlsOrBoysForSummer = 15.00
overnightPriceMixedGroupForWinter = 10.00
overnightPriceMixedGroupForSpring = 9.50
overnightPriceMixedGroupForSummer = 20.00

# спортовете, които ще се практикуват в зависимост от вида на ваканцията и групата:
girlsWinter = "Gymnastics"
girlsSpring = "Athletics"
girlsSummer = "Volleyball"
boysWinter = "Judo"
boysSpring = "Tennis"
boysSummer = "Football"
mixedGroupWinter = "Ski"
mixedGroupSpring = "Cycling"
mixedGroupSummer = "Swimming"

season = input()
group = input()
studentsCount = int(input())
overnightCount = int(input())

price = studentsCount * overnightCount
sport = ""

if group == 'boys':
    if season == 'Winter':
        sport = boysWinter
        price *= overnightPriceGirlsOrBoysForWinter
    elif season == 'Spring':
        sport = boysSpring
        price *= overnightPriceGirlsOrBoysForSpring
    elif season == 'Summer':
        sport = boysSummer
        price *= overnightPriceGirlsOrBoysForSummer
elif group == 'girls':
    if season == 'Winter':
        sport = girlsWinter
        price *= overnightPriceGirlsOrBoysForWinter
    elif season == 'Spring':
        sport = girlsSpring
        price *= overnightPriceGirlsOrBoysForSpring
    elif season == 'Summer':
        sport = girlsSummer
        price *= overnightPriceGirlsOrBoysForSummer
elif group == 'mixed':
    if season == 'Winter':
        sport = mixedGroupWinter
        price *= overnightPriceMixedGroupForWinter
    elif season == 'Spring':
        sport = mixedGroupSpring
        price *= overnightPriceMixedGroupForSpring
    elif season == 'Summer':
        sport = mixedGroupSummer
        price *= overnightPriceMixedGroupForSummer
    
#  • Ако броят на учениците е 50 или повече, училището получава 50% отстъпка
#  • Ако броят на учениците е 20 или повече и в същото време по-малък от 50, училището получава 15% отстъпка
#  • Ако броят на учениците е 10 или повече и в същото време по-малък от 20, училището получава 5% отстъпка
if studentsCount >= 50:
    price *= 0.5
elif studentsCount >= 20:
    price *= 0.85
elif studentsCount >= 10:
    price *= 0.95
    
print(f'{sport} {price:.2f} lv.')
